core/src/main/java/org/apache/calcite/rel/rel2sql/SqlImplementor.java

```diff
@@ -1119,6 +1119,34 @@ public List<SqlNode> fieldList() {
       };
     }
 
+    /**
+     * Checks if a generated ORDER BY node is a literal value.
+     * @param node to check
+     * @return true if node is representing a literal value, false otherwise.
+     */
+    boolean isLiteral(SqlNode node) {
```

Comment:
method should be private, or name of method should indicate that tests whether a node is a literal for purposes of the ORDER BY clause